### ISSUE TITLE
When migrating from 3.x to 4.x there is problem with case sensitivity in collection type attributes

### ISSUE THREAD

**Issue by sarsonj** [strapi/strapi#12636]
### Describe the bug
I am migrating strapi from 3.x to latest 4.1.0. I updated app using `codemods`, this updates models in API. However, in 3.x version, when I created field using content type manager with first uppercase letter (like Title), it also created same field in database (MySql in my case - Title). But in version 4.x, it looks like you force attribute name to lower cammelCase. 

What happens when I install Strapi 4 on existing database, it fails with error that table cannot be altered, because field already exists there (there is Title and Strapi is trying to create title).

### Steps to reproduce the behaviour

1. use strapi 3 with MySql and some attributes in collection type starting with upper case letter
2. migrate to strapi 4 using codemods
3. start strapi 4
4. it fails with Alter table error

**Comment by derrickmehaffy** [comment]
We aren't supported v3 to v4 migrations currently as we haven't completed the currently WIP migration guide.

You are welcome to attempt migration before then but we are not accepting issues on it at this time.

TLDR: You do it at your own risk
